pkg/controllers/workspace_controller.go

```diff
@@ -549,7 +649,8 @@ func (c *WorkspaceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&kaitov1alpha1.Workspace{}).
 		Owns(&appsv1.Deployment{}).
 		Owns(&appsv1.StatefulSet{}).
-		Watches(&v1alpha5.Machine{}, c.watchMachines()).
+		Watches(&v1alpha5.Machine{}, c.watchMachines()).    // watches for machine with labels indicating workspace name.
+		Watches(&v1beta1.NodeClaim{}, c.watchNodeClaims()). // watches for nodeClaim with labels indicating workspace name.
```

Comment:
Guard with feature gate. The watch call requires the nodeclaim CRD to be installed, otherwise, it will throw error.

Reply:
Addressed